### PR TITLE
adding decimal type support

### PR DESCRIPTION
**Pull request by zhexuany**
Since original aggregate function PR is too large which is not friendly for code review. Hence, I split that PR into two. This is the result of it.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tikv-client-java/33)

### CODE REVIEW

**Comment by zhexuany** [comment]
@shenli @ilovesoup is pretty busy now. And this PR is kind of emergency. Future works deadens on this PR. If you have time, PTAL.

File: src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import com.pingcap.tikv.codec.MyDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+public class MyDecimalTest {
+    @Test
+    public void fromStringTest() throws Exception {
+        List<MyDecimalTestStruct> test = new ArrayList<>();
+        test.add(new MyDecimalTestStruct("12345", "12345", 5, 0));
```
Comment (shenli) [inline src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java:25]: Miss these: 
https://github.com/pingcap/tidb/blob/master/util/types/mydecimal_test.go#L308
https://github.com/pingcap/tidb/blob/master/util/types/mydecimal_test.go#L315

File: src/test/java/com/pingcap/tikv/codec/DecimalUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.pingcap.tikv.codec;
```
Comment (shenli) [inline src/test/java/com/pingcap/tikv/codec/DecimalUtilsTest.java:14]: Miss license.

File: src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package com.pingcap.tikv.codec;
```
Comment (shenli) [inline src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java:14]: Miss license.

File: .travis.yml [hunk 4]
```diff
@@ -1,51 +1,9 @@
-dist: trusty
-sudo: required
 language: java
-os:
-  - linux
-
-env:
-  - V=0.4.5
-
-before_install:
```
Comment (shenli) [inline .travis.yml:10]: Why remove this?

File: src/main/java/com/pingcap/tikv/codec/DecimalUtils.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Arrays;
+import java.io.IOException;
+import java.lang.Byte;
+import javax.sound.midi.SysexMessage;
+import java.util.ArrayList;
+import java.util.List;
+import com.pingcap.tikv.codec.MyDecimal;
+import com.google.common.primitives.Ints;
+
+public class DecimalUtils {
```
Comment (shenli) [inline src/main/java/com/pingcap/tikv/codec/DecimalUtils.java:28]: Add comment for the two functions.

File: src/main/java/com/pingcap/tikv/codec/MyDecimal.java [hunk 6]
```diff
@@ -0,0 +1,755 @@
...
+    // The following are fields of MyDecimal
+    private int digitsInt;
+    private int digitsFrac;
+    private int resultFrac;
+    private boolean negative;
+    private int[] wordBuf = new int[maxWordBufLen];
+
+    /*
+     * Returns total precision of this decimal. Basically, it is sum of digitsInt and digitsFrac. But there
+     * are some special cases need to be takedn care of such as 000.001.
```
Comment (shenli) [inline src/main/java/com/pingcap/tikv/codec/MyDecimal.java]: takedn -> token

File: src/main/java/com/pingcap/tikv/codec/MyDecimal.java [hunk 7]
```diff
@@ -0,0 +1,755 @@
...
+            if (!Character.isSpaceChar(str[startIdx])) {
+                break;
+            }
+        }
+
+        if (str.length == 0) {
+            throw new IllegalArgumentException("BadNumber");
+        }
+
+        // skip sing and record where digits start from
```
Comment (shenli) [inline src/main/java/com/pingcap/tikv/codec/MyDecimal.java]: skip sing? sign?

File: .travis.yml (hunk 4 above)
Comment (zhexuany) [reply .travis.yml:10]: We plan use `bazel` in this project. That is why we change `.travis.yml` at first place. After changes, travis becomes much slower. When we use `mvn`, it only takes one or two minute. But `bazel`needs 4 minutes. The reason could be using `bazel` requires `sudo`.

**Comment by shenli** [comment]
Rest LGTM

File: src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java (hunk 1 above)
Comment (zhexuany) [reply src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java:25]: "12345." is valid, but "1245E5" is not valid in java.

**Review by shenli — APPROVED** [review]

File: src/main/java/com/pingcap/tikv/codec/DecimalUtils.java (hunk 5 above)
Comment (zhexuany) [reply src/main/java/com/pingcap/tikv/codec/DecimalUtils.java:28]: DONE

File: src/test/java/com/pingcap/tikv/codec/DecimalUtilsTest.java (hunk 2 above)
Comment (zhexuany) [reply src/test/java/com/pingcap/tikv/codec/DecimalUtilsTest.java:14]: DONE

File: src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java (hunk 3 above)
Comment (zhexuany) [reply src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java:14]: DONE

File: src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java (hunk 1 above)
Comment (zhexuany) [reply src/test/java/com/pingcap/tikv/codec/MyDecimalTest.java:25]: DONE.

**Pull request merged**